### ISSUE TITLE
Separate out 'unsafe' code in coverage reports for Rust projects

### ISSUE THREAD

**Issue by jrmuizel** [google/oss-fuzz#5660]
In addition to Line Coverage, Function Coverage and Region Coverage, it would be useful to have Unsafe Coverage. This would give a could overall sense of the coverage in dangerous areas.

I have no idea how practical this would be to implement.

**Comment by jonathanmetzman** [comment]
It is an interesting idea. I would guess we maintainers don't have the cycles for this, but it would be interesting to have so maybe we can leave this open.
Do you have an example project where this would be helpful?
I one time went through half-a-dozen random rust projects here and searched their code, none had unsafe.

I think I looked this up because I was thinking about teaching libFuzzer to target unsafe code. Would be interested in hearing thoughts from rust folks on this too.

**Comment by jrmuizel** [comment]
The project that I had in mind was https://github.com/FirefoxGraphics/qcms

Having the ability for libFuzzer target unsafe code sounds pretty useful. Do you have a rough idea for how that would work?

**Comment by jrmuizel** [comment]
> Sorry for late follow up.
> 
> > The project that I had in mind was https://github.com/FirefoxGraphics/qcms
> 
> But does that have any unsafe code? Sorry I know nothing about rust and [searching](https://github.com/FirefoxGraphics/qcms/search?q=unsafe) turns up nothing.

Yeah there's a bunch here: https://github.com/FirefoxGraphics/qcms/blob/master/src/transform.rs (searching forked repos doesn't search the code)

**Comment by jonathanmetzman** [comment]
> > Sorry for late follow up.
> > > The project that I had in mind was https://github.com/FirefoxGraphics/qcms
> > 
> > 
> > But does that have any unsafe code? Sorry I know nothing about rust and [searching](https://github.com/FirefoxGraphics/qcms/search?q=unsafe) turns up nothing.
> 
> Yeah there's a bunch here: https://github.com/FirefoxGraphics/qcms/blob/master/src/transform.rs (searching forked repos doesn't search the code)

Lol I guessed that was probably the issue. Thanks!
